docs/request-custom.md

```diff
@@ -0,0 +1,210 @@
+                response?.data ?: ByteArray(0),
+                Charset.forName(HttpHeaderParser.parseCharset(response?.headers)))
+        Response.success(
+                gson.fromJson(json, clazz),
+                HttpHeaderParser.parseCacheHeaders(response))
+    }
+    // handle errors
+}
+```
+
+*Java*
+
+```java
+@Override
+protected Response<T> parseNetworkResponse(
+        NetworkResponse response) {
+    try {
+        String json = new String(response.data,
+        HttpHeaderParser.parseCharset(response.headers));
+    return Response.success(gson.fromJson(json, clazz),
+    HttpHeaderParser.parseCacheHeaders(response));
+    }
+    // handle errors
+}
+```
+
+Note the following:
+
+- `parseNetworkResponse()` takes as its parameter a `NetworkResponse`, which
+  contains the response payload as a byte[], HTTP status code, and response headers.
+- Your implementation must return a `Response<T>`, which contains your typed
+  response object and cache metadata or an error, such as in the case of a parse failure.
+
+If your protocol has non-standard cache semantics, you can build a `Cache.Entry`
+yourself, but most requests are fine with something like this:
+
+*Kotlin*
+
+```kotlin
+return Response.success(myDecodedObject,
+        HttpHeaderParser.parseCacheHeaders(response))
+```
+
+*Java*
+
+```java
+return Response.success(myDecodedObject,
+        HttpHeaderParser.parseCacheHeaders(response));
+```
+
+Volley calls `parseNetworkResponse()` from a worker thread. This ensures that
+expensive parsing operations, such as decoding a JPEG into a Bitmap, don't block the UI
+thread.
+
+### deliverResponse
+
+Volley calls you back on the main thread with the object you returned in
+`parseNetworkResponse()`. Most requests invoke a callback interface here,
+for example:
+
+*Kotlin*
+
+```kotlin
+override fun deliverResponse(response: T) = listener.onResponse(response)
+```
+
+*Java*
+
+```java
+protected void deliverResponse(T response) {
+    listener.onResponse(response);
+}
+```
+
+## Example: GsonRequest
+
+[Gson](https://github.com/google/gson) is a library for converting
+Java objects to and from JSON using reflection. You can define Java objects that have the
+same names as their corresponding JSON keys, pass Gson the class object, and Gson will fill
+in the fields for you. Here's a complete implementation of a Volley request that uses
+Gson for parsing:
+
+*Kotlin*
+
+```kotlin
+/**
+ * Make a GET request and return a parsed object from JSON.
+ *
+ * @param url URL of the request to make
+ * @param clazz Relevant class object, for Gson's reflection
+ * @param headers Map of request headers
+ */
+class GsonRequest<T>(
+        url: String,
+        private val clazz: Class<T>,
+        private val headers: MutableMap<String, String>?,
+        private val listener: Response.Listener<T>,
+        errorListener: Response.ErrorListener
+) : Request<T>(Method.GET, url, errorListener) {
+    private val gson = Gson()
+
```

Comment:
nit: I see this is carried over, but the extra blank line here seems unnecessary